core/src/main/kotlin/fr/sncf/osrd/stdcm/infra_exploration/InfraExplorer.kt

```diff
@@ -286,19 +293,36 @@ private class InfraExplorerImpl(
                 val isRouteBlocked =
                     constraints.any { constraint -> constraint.apply(block).isNotEmpty() }
                 if (isRouteBlocked) return false
-                val endPath = endBlocks.contains(block)
+                val endLocationsOnBlock =
+                    if (stops.isEmpty()) {
+                        listOf()
```

Comment:
It feels like this should be impossible. Has this case happened during testing?

I don't think we would find any valid path in this case, we could just `return false` to avoid the pathfinding process entirely

Reply:
It's indeed happening during tests (17 cases).
Returning `false` also crashes, so it's debatable, but I'd keep it like that as it doesn't degrade things.

Reply:
Oh I forgot that those aren't just waypoints but specifically points where the train stops.

This means that the last waypoint isn't always a stop. This was supposed to be enforced in the endpoint, but maybe this case happens in unit tests? 

The problem here is that we use the last *stop* to define when the path is marked as complete, when we should be using the last waypoint. This will cause bugs, the kind that may go unnoticed.

The easiest way to handle this would be to *ensure* that the last waypoint is a stop, including in test (though either asserts or by setting it by hand). The other solution would be to keep the previous parameter (`endBlocks`) on top of the stops.

Reply:
As discussed, we dug a cleaner solution in https://github.com/OpenRailAssociation/osrd/pull/7322/commits/d56a4c1a43aaf050ee07f9c47a1e36b34cd1cca3